app/views/unattended/provisioning_templates/snippet/remote_execution_ssh_keys_kickstart.erb

```diff
@@ -0,0 +1,59 @@
+<%#
+kind: snippet
+name: remote_execution_ssh_keys_kickstart
+model: ProvisioningTemplate
+snippet: true
+-%>
+<%#
+# SSH keys setup snippet for Remote Execution plugin via Kickstart format
+#
+# Parameters:
+#
+# remote_execution_ssh_keys: public keys to be put in ~/.ssh/authorized_keys
+#
+# remote_execution_ssh_user: user for which remote_execution_ssh_keys will be
+#                            authorized
+#
+# remote_execution_create_user: create user if it not already existing
+#
+# remote_execution_effective_user_method: method to switch from ssh user to
+#                                         effective user
+#
+# This template sets up SSH keys in any host so that as long as your public
+# SSH key is in remote_execution_ssh_keys, you can SSH into a host. This
+# works in combination with Remote Execution plugin by querying smart proxies
+# to build an array.
+#
+# To use this snippet without the plugin provide the SSH keys as host parameter
+# remote_execution_ssh_keys. It expects the same format like the authorized_keys
+# file.
+#
+# This snippet must be included outside of a %post section.
+#
+-%>
+<% if host_param_true?('host_registration_remote_execution') && !host_param('remote_execution_ssh_keys').blank? -%>
+<% ssh_user = host_param('remote_execution_ssh_user') || 'root' %>
+
+<% if ssh_user != 'root' && host_param_true?('remote_execution_create_user') -%>
+user --name=<%= ssh_user %>
+<% end -%>
+
+<% if host_param('remote_execution_ssh_keys').responds_to?(:each) -%>
+<% host_param('remote_execution_ssh_keys').each do |key| -%>
+sshkey --username=<%= ssh_user %> "<%= key %>"
+<% end -%>
+<% else -%>
+sshkey --username=<%= ssh_user %> "<%= host_param('remote_execution_ssh_keys') %>"
```

Comment:
The parameter might be a string, containing multiple newline separated public keys. Is `sshkey` able to deal with it?

Reply:
No it must be a single key, this was not clear from the documentation in the original snippet. I thought it's an array. Will modify than, what else can I expect?

Reply:
Can I just assume this is a string and call `split` on it? Why the original has "if it's a String do something else"?

Reply:
You do need to split on lines, not just any whitespace. The user can input this as a host parameter if the plugin is not installed.

Reply:
Also in theory if the parameter is an array, each element can be a string containing multiple newline-delimited public keys. So maybe wrapping the parameter value in an array, flattening it and the iterating over it and splitting each value by newlines would be the safest approach.